encryption/classic-mceliece/classic-mceliece-kem/classic-mceliece-kem.yaml

```diff
@@ -0,0 +1,4 @@
+name: Classic McEliece
+type: KEM
+security notion: IND-CCA
+security notion comment: CCA2
```

Comment:
> In general, very good! :)
> 
> But still some things to improve, some remarks below. See also the comments inline.
> 
> I think the PKE is not really a submission to the contest, a) because they don't mention it is and b) because it does not comply with the api (its input is a weight-_t_ vector, not a bitstring). Also I think the numbers in table 1 are for the KEM only. So I'd say we cannot really keep the PKE. On the other hand, one could argue that the FPGA timings are better placed under pke (because they don't include the hashing)

Before the PR, I asked Benny about it. He said it should be part of Classic McEliece, that is why it is in.    

> Where are the bit security values from? I don't find them in the paper, did you maybe forget to include some source? Or did I overlook something?

There aren't explicit values. 
"Security level 2^b thus uses key size (C0+o(1))b^2(log(b))^2 where C0=R/(1−R)(lg(1−R))2."(page 32, https://classic.mceliece.org/nist/mceliece-20190331.pdf).
I calculated the bit security, they could be a bit vague because of R. I choosed R = 0.79681213 and calculated C0, which was close to 1.5. 
But after reading the page 32/33, again I have some second thoughts about it. "This C0 reaches its minimum value, approximately 0.7418860694, when R is approximately 0.7968121300."

Reply:
@benj-zen what do you say about the PKE thing?

Reply:
OK I really don't get these calculations. One _might_ deduce from their writing that the parameters can be fine-tuned that well that they meet 128/192/256 bits exactly, but I'm not sure about this either. (And still, if so, we don't have quantum security numbers.) Shall we ask the authors maybe?

Reply:
I'm against including the PKE. Most of the submissions are based on a PKE using the underlying problem. The KEM-transformation and the security proofs are then (most of the time) straightforward. I don't think McEliece is an exception here.
Problem is that we don't have any benchmarks, most of the time no specific implementation or api, and not even really parameter sets.